### ISSUE TITLE
DataViewWidget should not allow data_model to be None or handle that case

### ISSUE THREAD

**Issue by corranwebster** [enthought/pyface#805]
At least at the point where `_create()` is called, the `data_model` should be non-`None`, otherwise we get errors with basic operations like sizing the view.  Should possibly consider just having `allow_none=False` for the `data_model` trait.

Alternatively, we could make the underlying Qt model robust to missing data models if we can provide sensible defaults.

**Comment by mdickinson** [comment]
> Should possibly consider just having `allow_none=False` for the `data_model` trait.

I'm not sure that will help much. The most likely reason for `data_model` being `None` would be that the `data_model` simply hasn't been assigned a value, so we're picking up the `None` default. Using `allow_none=False` will only catch the (I'd guess less common) case where someone's deliberately assigning a value of `None` for the `data_model`.

**Comment by corranwebster** [comment]
> I'm not sure that will help much. The most likely reason for `data_model` being `None` would be that the `data_model` simply hasn't been assigned a value, so we're picking up the `None` default.

Right.  It would need to be coupled with `HasRequiredTraits` here: https://github.com/enthought/pyface/blob/e4e4b3303fdcb0ee002d6414a49a2240d3e185e0/pyface/data_view/i_data_view_widget.py#L62
